src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -477,6 +479,13 @@ public void RecordRunning(Message message)
             // Note: This method is always called while holding lock on this activation, so no need for additional locks here
 
             numRunning++;
+            if (message.Direction != Message.Directions.OneWay 
+                && message.SendingGrain != null
+                && !message.SendingGrain.IsClient)
+            {
+                RunningRequestsSenders.Add(message.SendingGrain);
```

Comment:
I think this should be `SendingActivation`, instead. It's a corner case, but we could see multiple activations with the same `GrainId` (i.e, during partial failure scenarios).